fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/rfc/Generator.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.apirequest.rfc;
+
+import com.yahoo.bard.webservice.application.AbstractBinderFactory;
+import com.yahoo.bard.webservice.web.apirequest.DataApiRequest;
+import com.yahoo.bard.webservice.web.util.BardConfigResources;
+
+/**
+ * Contract for a builder of a piece of a {@link com.yahoo.bard.webservice.web.apirequest.DataApiRequest}. Each
+ * generator has a bind and validate step, where the resource is generated and then validated being added to the
+ * in progress api request.
+ *
+ * <p>To modify how a data api request is built, you must implement this interface for the type of resource you want
+ * to generate. In your {@link AbstractBinderFactory} implementation you must make a named binding for the type of
+ * resource you wish to generate. You can only bind one generator per resource (though different resources may share
+ * a type, such as "count" and "topN"), so your implementation will replace the default generator for that resource.
+ *
+ * TODO: binding names should be written here for reference.
+ * TODO: binding order should be written here
+ *
+ * @param <T> The type of the resource this generator builds.
+ */
+public interface Generator<T> {
```

Comment:
This interface probably deserves a slightly more specific name.  I'm not thinking of one yet, though, so this is unhelpful feedback at the moment.